hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -390,13 +410,9 @@ func (r *NodePoolReconciler) reconcileInPlaceUpgradeManifests(ctx context.Contex
 	}
 
 	configmap := inPlaceUpgradeConfigMap(nodePool, namespace.Name)
-
 	if result, err := r.CreateOrUpdate(ctx, hostedClusterClient, configmap, func() error {
 		return r.reconcileUpgradeConfigmap(
-			ctx,
-			configmap,
-			targetConfigVersionHash, ignEndpoint,
-			caCertBytes, tokenBytes,
+			ctx, configmap, targetConfigVersionHash, payload,
```

Comment:
By the time we get here, is the token secret guaranteed up to date? The contents are processed via tokensecret_controller, so would it be possible that the update hasn't happened?

Reply:
its guaranteed by https://github.com/openshift/hypershift/pull/1290/files#diff-186fca173b913fab90bc6a26071e1255532f14ce6156a796e04de919d8e97c13R64

If the version or config changes meanwhile a new secret is created and a new upgrade process towards a new versionConfig is triggered.

Reply:
Ah gotcha, thanks!